### PR TITLE
ARM: dts: qcom: apq8026-samsung-milletwifi: Add touchscreen buttons

### PR DESCRIPTION
**Pull request by Susurrus**

### CODE REVIEW

**Comment by z3ntu** [comment]
Signed-off-by please in the commit message, and add some comment - or indicate if it should be squashed into some other commit next time I rebase.

**Comment by Susurrus** [comment]
Sorry about that. Didn't realize that's how this should work. I think I annotated this correctly now.

Also, 432ab7333549a41e9f6288d60aeab55bf51d7f3e is a hack, does that also need an annotation? Probably should open an issue here to track it as there's some other context around it I have in my notes too.

**Comment by Susurrus** [comment]
Note that `matisse-wifi` uses `APPSELECT`, so I'm confused which one should be used here. Is there any documentation I can reference here?

**Comment by z3ntu** [comment]
I'd check what's more commonly used on phones in mainline. Not sure there's docs for input like this, in a similar area there is some docs for the "function" for leds so notification led is defined rather well

**Comment by Susurrus** [comment]
`MENU` is way more common. However, I found https://gitlab.com/postmarketOS/pmaports/-/commit/c655f5cc9c4110d4ff118327c52c5c1d4841eeb5 which is weird, that the DTS should have `APPSELECT` but then pmOS should remap it?

I'm wondering if I should open a general pmOS issue on this or maybe ask in the GNOME Discourse if they have any opinion. WDYT?

**Comment by z3ntu** [comment]
You could for sure ask in the postmarketOS mainline channel what other people think. My devices don't have these touchkeys (well, apart from OnePlus One I believe but there the touchscreen driver doesn't support the keys yet) so I cannot tell you much there myself.

**Comment by Susurrus** [comment]
Since I'm using what's most common right now, it's probably fine for the initial bringup work. Any objections from you to merging this as-is?

File: arch/arm/boot/dts/qcom/qcom-apq8026-samsung-milletwifi.dts [hunk 1]
```diff
@@ -304,6 +304,7 @@
 		touchscreen-size-y = <1280>;
 		avdd-supply = <&reg_tsp_3p3v>;
 		vdd-supply = <&reg_tsp_1p8v>;
+		linux,keycodes = <KEY_MENU KEY_BACK>;
```
Comment (z3ntu) [inline arch/arm/boot/dts/qcom/qcom-apq8026-samsung-milletwifi.dts]: Based on your messages on Matrix I think you said it should be `KEY_APPSELECT` for this device, no?

**Comment by Susurrus** [comment]
Correct, but I want to test it again before resubmitting.

**Comment by Susurrus** [comment]
Tested and working. Included in my latest round of upstreaming patches.

**Comment by z3ntu** [comment]
> Tested and working. Included in my latest round of upstreaming patches.

Are you sure? Your v3 on the mailing lists uses this line?
```c
linux,keycodes = <KEY_APPSELECT KEY_BACK>;
```
https://lore.kernel.org/linux-arm-msm/20240219214643.197116-3-bryant@mai.rs/

**Comment by Susurrus** [comment]
Whoops, looks like I didn't force-push the changes but had them locally.

However, now that you have a 6.8 branch, should I instead merge this there?

**Comment by z3ntu** [comment]
Pushed https://github.com/msm8226-mainline/linux/commit/b73ea5a88849d2849056f936e7946e695957c197 to the 6.8 branch now, that should resolve this PR :)

**Pull request closed without merging**